src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -258,6 +259,40 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
                 .put("compatibility_mode", eventType.getCompatibilityMode()));
     }
 
+    private Multimap<TopicRepository, String> deleteEventTypeIfNoSubscriptions(final String eventType) {
+        if (hasSubscriptions(eventType)) {
+            throw new ConflictException("Can't remove event type " + eventType
+                    + ", as it has subscriptions");
+        }
+        return transactionTemplate.execute(action -> deleteEventType(eventType));
```

Comment:
Do you need to do it in a transaction?

Reply:
It used to run inside a transaction, so I didn't change anything here. 
I think it is needed because delete event-type also handles the deletion of timelines.